### PR TITLE
zfsUnstable: 2.1.13 -> 2.2.0-rc4

### PR DESCRIPTION
**Pull request by Mic92**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Still breaking the normal zfs build. I will have a look.

**Comment by adamcstephens** [comment]
There is some discussion in duplicate #245330 about waiting until 2.2.0 is released before updating zfsUnstable.

**Comment by RaitoBezarius** [comment]
I chatted about it with Mic92 in private FWIW.

**Review by yu-re-ka — COMMENTED** [review]
zfs 2.2.0 will also add armv7l to the cpu targets block; please add `isAarch32` in meta.platforms accordingly

**Comment by RaitoBezarius** [comment]
I would prefer we go for https://github.com/NixOS/nixpkgs/pull/246300 first, then we can consider this open until 2.2.0-rcX is run enough by our maintainers I'd say?

**Comment by Mic92** [comment]
Works well so far on my machine.

**Comment by Mic92** [comment]
Updated to latest rc, which also fixes https://github.com/openzfs/zfs/issues/15140

**Comment by adamcstephens** [comment]
Commit and PR title are both incorrect. 2.1.13 and 2.2.0-rc4

I've applied the kernel module to one of my systems, but getting a failure from `extra-utils` when trying to overlay the user space tools. Unsure if that is related to this PR or not, and is not a helpful failure.

```
patching /nix/store/1vkvigd0ggmx83gqrr5swnpq383g7wdn-extra-utils/lib/libdevmapper.so.1.02...
patching /nix/store/1vkvigd0ggmx83gqrr5swnpq383g7wdn-extra-utils/lib/libbrotlidec.so.1...
testing patched programs...
hello world
253
mdadm - v4.2 - 2021-12-30
```

**Comment by RaitoBezarius** [comment]
I think the PR should allow 6.5 kernel or did I miss something?

**Comment by RaitoBezarius** [comment]
@ofborg build zfs

**Comment by adamcstephens** [comment]
Yes, this is still failing to build stage1 for me.

```
extra-utils> ++ /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs --help
extra-utils> + exitHandler
extra-utils> + exitCode=127
extra-utils> + set +e
```

```
─❯ /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs --help
/nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs: symbol lookup error: /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs: undefined symbol: sa_protocol_names
```

**Comment by Mic92** [comment]
Interesting. I am using 

> Yes, this is still failing to build stage1 for me.
> 
> ```
> extra-utils> ++ /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs --help
> extra-utils> + exitHandler
> extra-utils> + exitCode=127
> extra-utils> + set +e
> ```
> 
> ```
> ─❯ /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs --help
> /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs: symbol lookup error: /nix/store/icqhklalv093a6psh6hiv7c8dqlwa9wd-extra-utils/bin/zfs: undefined symbol: sa_protocol_names
> ```

Cannot reproduce this issue. I found an error in the test, that I fixed here: https://github.com/NixOS/nixpkgs/pull/255484

**Comment by adamcstephens** [comment]
Ok, I narrowed it down to the `zfsutil` filesystem option when used on a filesystem needed for boot (e.g. /, /nix).

**Comment by Mic92** [comment]
Found the issue:

In extra-utils for the initrd we do: `stripDirs "$STRIP" "$RANLIB" "lib bin" "-s"`
This is causing that the `sa_protocol_names` gets dropped from the shared object.

**Comment by Mic92** [comment]
Update this is not the actual issue. But here is the fix: https://github.com/NixOS/nixpkgs/pull/255583

**Comment by enterJazz** [comment]
> I think the PR should allow 6.5 kernel or did I miss something? EDIT: actually no :-(

How difficult would it be to support 6.5? @RaitoBezarius

File: pkgs/os-specific/linux/zfs/unstable.nix [hunk 1]
```diff
@@ -16,17 +16,16 @@ callPackage ./generic.nix args {
     else kernel.kernelOlder "6.2";
 
   latestCompatibleLinuxPackages = if stdenv'.isx86_64 || removeLinuxDRM
-    then linuxKernel.packages.linux_6_4
-    else linuxKernel.packages.linux_6_1;
+  then linuxKernel.packages.linux_6_4
+  else linuxKernel.packages.linux_6_1;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/zfs/unstable.nix:20]: This is inconsistent with the rest of the file

Comment (adamcstephens) [reply pkgs/os-specific/linux/zfs/unstable.nix:20]: Why are you nitpicking a formatting inconsistency 3 weeks after a PR was merged?

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/zfs/unstable.nix:20]: Because I only stumbled upon this PR yesterday when I dug up my old zfsUnstable bump PR.

**Pull request merged**